File: Src/OnlineExam/ClientApp/src/app/common/baseResponses/baseResponse.ts
import { expressionType } from '@angular/compiler/src/output/output_ast';
import { JsonObject, JsonProperty } from 'typescript-json-serializer';

export enum TypeResponse {
  Success,
  BusinessException,
  ErrorException,
}

@JsonObject()
export class ErrorResponse {
  public Type?: string;
  public Message?: string;
}

@JsonObject()
export class BaseResponse<T> {
  @JsonProperty() public Type?:number;
  @JsonProperty() public data?: T;
  @JsonProperty() public type?: TypeResponse;
  @JsonProperty({ type: ErrorResponse }) public error?: ErrorResponse;
}

File: Src/OnlineExam/Common/Models/BaseResponse.cs
using System.Text.Json.Serialization;

namespace OnlineExam.Common.Models;

public class BaseResponse
{
    [JsonPropertyName("type")]
    public TypeResponse Type { get; set; }
    [JsonPropertyName("data")]
    public object Data { get; set; }
    [JsonPropertyName("error")]
    public ErrorResponse Error { get; set; }
    [JsonPropertyName("errorException")]
    public ErrorResponse ErrorException { get; set; }
}